lib/components/steps/Home.tsx

```diff
@@ -1,42 +1,36 @@
-import { Box, Button, Typography, Chip } from '@mui/material';
-import { useCdeContext } from "../../CdeContext.tsx";
-import { STEPS } from "../../models.ts";
-import { validateInputMappings, processInputMappings } from '../../services/mappingService.ts';
-import { ModalLayout } from "../layout/ModalLayout.tsx";
-import { StyledTable } from '../common/StyledTable.tsx';
-import { CircleChipDefault, CircleChipSuccess } from '../../icons/index.tsx';
-import { vars } from '../../theme/variables.ts';
-
-const { primary600, gray600 } = vars;
+import {Box, Button, Typography, Chip} from '@mui/material';
+import {useCdeContext} from "../../CdeContext.tsx";
+import {STEPS} from "../../models.ts";
+import {areMappingsValid, getMappingsDict} from '../../services/mappingService.ts';
+import {ModalLayout} from "../layout/ModalLayout.tsx";
+import {StyledTable} from '../common/StyledTable.tsx';
+import {CircleChipDefault, CircleChipSuccess} from '../../icons/index.tsx';
+import {vars} from '../../theme/variables.ts';
+const {primary600, gray600} = vars;
 
 function Home() {
     const {
+        labName,
         setStep,
-        inputMappings,
+        mappings,
+        datasetSample,
         setLoadingMessage,
         setErrorMessage,
-        setMapping
+        setCDEMapping
     } = useCdeContext();
 
     const handleStartMapping = async () => {
-        if (inputMappings.length == 0) {
-            setErrorMessage('Mapping file not found')
-            return
-        }
-
         setLoadingMessage('Validating and processing file...');
 
         try {
-            const isValid = validateInputMappings(inputMappings);
-            if (isValid) {
-                const mapping = processInputMappings(inputMappings);
-                setMapping(mapping);
+            if (areMappingsValid(mappings)) {
+                const mappingsDict = getMappingsDict(mappings)
```

Comment:
@Aiga115 likewise

Reply:
Question addressed in https://github.com/MetaCell/cde-mapper/pull/12/files/ab7021052fb67f33d62e6695ef249dbc98fddbb9#r1452609886